### ISSUE TITLE
[PATCH] Upgrade to Thread::Queue 3.12

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#15867]
Migrated from [rt.perl.org#130752](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=130752) (status was 'rejected')

Searchable as RT130752$

**Comment by p5pRT** [comment]
### From @jdhedden

This is a bug report for perl from jdhedden@&#8203;cpan\.org\,
generated with the help of perlbug 1\.40 running under perl 5\.24\.1\.

The attached patch upgrades 'Thread&#8203;::Queue' to version 3\.12\.

This is a fix from Chad Fox for
\[rt\.cpan\.org \#120157\] deadlock using dequeue\_nb on size limited queue
https://rt.cpan.org/Ticket/Display.html?id=120157


---
Flags&#8203;:
&nbsp;    category=library
&nbsp;    severity=low
&nbsp;    Type=Patch
&nbsp;    PatchStatus=HasPatch
&nbsp;    module=Thread&#8203;::Queue


---
Site configuration information for perl 5\.24\.1&#8203;:

Configured by Debian Project at Sun Jan 15 23&#8203;:35&#8203;:20 UTC 2017\.

Summary of my perl5 \(revision 5 version 24 subversion 1\) configuration&#8203;:
&nbsp;   
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=3\.16\.0\, archname=x86\_64\-linux\-gnu\-thread\-multi
&nbsp;    uname='linux localhost 3\.16\.0 \#1 smp debian 3\.16\.0 x86\_64 gnulinux '
&nbsp;    config\_args='\-Dusethreads \-Duselargefiles \-Dcc=x86\_64\-linux\-gnu\-gcc \-Dcpp=x86\_64\-linux\-gnu\-cpp \-Dld=x86\_64\-linux\-gnu\-gcc \-Dccflags=\-DDEBIAN \-Wdate\-time \-D\_FORTIFY\_SOURCE=2 \-g \-O2 \-fdebug\-prefix\-map=/build/perl\-eDFQz3/perl\-5\.24\.1=\. \-fstack\-protector\-strong \-Wformat \-Werror=format\-security \-Dldflags= \-Wl\,\-z\,relro \-Dlddlflags=\-shared \-Wl\,\-z\,relro \-Dcccdlflags=\-fPIC \-Darchname=x86\_64\-linux\-gnu \-Dprefix=/usr \-Dprivlib=/usr/share/perl/5\.24 \-Darchlib=/usr/lib/x86\_64\-linux\-gnu/perl/5\.24 \-Dvendorprefix=/usr \-Dvendorlib=/usr/share/perl5 \-Dvendorarch=/usr/lib/x86\_64\-linux\-gnu/perl5/5\.24 \-Dsiteprefix=/usr/local \-Dsitelib=/usr/local/share/perl/5\.24\.1 \-Dsitearch=/usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1 \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/share/man/man3 \-Dsiteman1dir=/usr/local/man/man1 \-Dsiteman3dir=/usr/local/man/man3 \-Dusesitecustomize \-Duse64bitint \-Dman1ext=1 \-Dman3ext=3perl
\-Dpager=/usr/bin/sensible\-pager \-Uafs \-Ud\_csh \-Ud\_ualarm \-Uusesfio \-Uusenm \-Ui\_libutil \-Uversiononly \-DDEBUGGING=\-g \-Doptimize=\-O2 \-dEs \-Duseshrplib \-Dlibperl=libperl\.so\.5\.24\.1'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='x86\_64\-linux\-gnu\-gcc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fwrapv \-fno\-strict\-aliasing \-pipe \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;    optimize='\-O2 \-g'\,
&nbsp;    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fwrapv \-fno\-strict\-aliasing \-pipe \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='6\.3\.0 20161229'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678\, doublekind=3
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16\, longdblkind=3
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='x86\_64\-linux\-gnu\-gcc'\, ldflags =' \-fstack\-protector\-strong \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib/gcc/x86\_64\-linux\-gnu/6/include\-fixed /usr/include/x86\_64\-linux\-gnu /usr/lib /lib/x86\_64\-linux\-gnu /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib
&nbsp;    libs=\-lgdbm \-lgdbm\_compat \-ldb \-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    perllibs=\-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    libc=libc\-2\.24\.so\, so=so\, useshrplib=true\, libperl=libperl\.so\.5\.24
&nbsp;    gnulibc\_version='2\.24'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib \-fstack\-protector\-strong'

Locally applied patches&#8203;:
&nbsp;    DEBPKG&#8203;:debian/cpan\_definstalldirs \- Provide a sensible INSTALLDIRS default for modules installed from CPAN\.
&nbsp;    DEBPKG&#8203;:debian/db\_file\_ver \- https://bugs.debian.org/340047 Remove overly restrictive DB\_File version check\.
&nbsp;    DEBPKG&#8203;:debian/doc\_info \- Replace generic man\(1\) instructions with Debian\-specific information\.
&nbsp;    DEBPKG&#8203;:debian/enc2xs\_inc \- https://bugs.debian.org/290336 Tweak enc2xs to follow symlinks and ignore missing @&#8203;INC directories\.
&nbsp;    DEBPKG&#8203;:debian/errno\_ver \- https://bugs.debian.org/343351 Remove Errno version check due to upgrade problems with long\-running processes\.
&nbsp;    DEBPKG&#8203;:debian/libperl\_embed\_doc \- https://bugs.debian.org/186778 Note that libperl\-dev package is required for embedded linking
&nbsp;    DEBPKG&#8203;:fixes/respect\_umask \- Respect umask during installation
&nbsp;    DEBPKG&#8203;:debian/writable\_site\_dirs \- Set umask approproately for site install directories
&nbsp;    DEBPKG&#8203;:debian/extutils\_set\_libperl\_path \- EU&#8203;:MM&#8203;: set location of libperl\.a under /usr/lib
&nbsp;    DEBPKG&#8203;:debian/no\_packlist\_perllocal \- Don't install \.packlist or perllocal\.pod for perl or vendor
&nbsp;    DEBPKG&#8203;:debian/fakeroot \- Postpone LD\_LIBRARY\_PATH evaluation to the binary targets\.
&nbsp;    DEBPKG&#8203;:debian/instmodsh\_doc \- Debian policy doesn't install \.packlist files for core or vendor\.
&nbsp;    DEBPKG&#8203;:debian/ld\_run\_path \- Remove standard libs from LD\_RUN\_PATH as per Debian policy\.
&nbsp;    DEBPKG&#8203;:debian/libnet\_config\_path \- Set location of libnet\.cfg to /etc/perl/Net as /usr may not be writable\.
&nbsp;    DEBPKG&#8203;:debian/mod\_paths \- Tweak @&#8203;INC ordering for Debian
&nbsp;    DEBPKG&#8203;:debian/prune\_libs \- https://bugs.debian.org/128355 Prune the list of libraries wanted to what we actually need\.
&nbsp;    DEBPKG&#8203;:fixes/net\_smtp\_docs \- \[rt\.cpan\.org \#36038\] https://bugs.debian.org/100195 Document the Net&#8203;::SMTP 'Port' option
&nbsp;    DEBPKG&#8203;:debian/perlivp \- https://bugs.debian.org/510895 Make perlivp skip include directories in /usr/local
&nbsp;    DEBPKG&#8203;:debian/deprecate\-with\-apt \- https://bugs.debian.org/747628 Point users to Debian packages of deprecated core modules
&nbsp;    DEBPKG&#8203;:debian/squelch\-locale\-warnings \- https://bugs.debian.org/508764 Squelch locale warnings in Debian package maintainer scripts
&nbsp;    DEBPKG&#8203;:debian/skip\-upstream\-git\-tests \- Skip tests specific to the upstream Git repository
&nbsp;    DEBPKG&#8203;:debian/patchlevel \- https://bugs.debian.org/567489 List packaged patches for 5\.24\.1\-1 in patchlevel\.h
&nbsp;    DEBPKG&#8203;:debian/skip\-kfreebsd\-crash \- https://bugs.debian.org/628493 \[perl \#96272\] Skip a crashing test case in t/op/threads\.t on GNU/kFreeBSD
&nbsp;    DEBPKG&#8203;:fixes/document\_makemaker\_ccflags \- https://bugs.debian.org/628522 \[rt\.cpan\.org \#68613\] Document that CCFLAGS should include $Config\{ccflags\}
&nbsp;    DEBPKG&#8203;:debian/find\_html2text \- https://bugs.debian.org/640479 Configure CPAN&#8203;::Distribution with correct name of html2text
&nbsp;    DEBPKG&#8203;:debian/perl5db\-x\-terminal\-emulator\.patch \- https://bugs.debian.org/668490 Invoke x\-terminal\-emulator rather than xterm in perl5db\.pl
&nbsp;    DEBPKG&#8203;:debian/cpan\-missing\-site\-dirs \- https://bugs.debian.org/688842 Fix CPAN&#8203;::FirstTime defaults with nonexisting site dirs if a parent is writable
&nbsp;    DEBPKG&#8203;:fixes/memoize\_storable\_nstore \- \[rt\.cpan\.org \#77790\] https://bugs.debian.org/587650 Memoize&#8203;::Storable&#8203;: respect 'nstore' option not respected
&nbsp;    DEBPKG&#8203;:debian/regen\-skip \- Skip a regeneration check in unrelated git repositories
&nbsp;    DEBPKG&#8203;:debian/makemaker\-pasthru \- https://bugs.debian.org/758471 Pass LD settings through to subdirectories
&nbsp;    DEBPKG&#8203;:debian/makemaker\-manext \- https://bugs.debian.org/247370 Make EU&#8203;::MakeMaker honour MANnEXT settings in generated manpage headers
&nbsp;    DEBPKG&#8203;:debian/devel\-ppport\-reproducibility \- https://bugs.debian.org/801523 Sort the list of XS code files when generating RealPPPort\.xs
&nbsp;    DEBPKG&#8203;:debian/encode\-unicode\-bom\-doc \- https://bugs.debian.org/798727 Document Debian backport of Encode&#8203;::Unicode fix
&nbsp;    DEBPKG&#8203;:debian/kfreebsd\-softupdates \- https://bugs.debian.org/796798 Work around Debian Bug\#796798
&nbsp;    DEBPKG&#8203;:fixes/autodie\-scope \- https://bugs.debian.org/798096 Fix a scoping issue with "no autodie" and the "system" sub
&nbsp;    DEBPKG&#8203;:fixes/crosscompile\-no\-targethost \- \[23695c0\] \[perl \#127234\] Fix the Configure escape with usecrosscompile but no targethost
&nbsp;    DEBPKG&#8203;:fixes/memoize\-pod \- \[rt\.cpan\.org \#89441\] Fix POD errors in Memoize
&nbsp;    DEBPKG&#8203;:fixes/ok\-pod \- Added encoding for pod\.
&nbsp;    DEBPKG&#8203;:debian/hurd\-softupdates \- https://bugs.debian.org/822735 Fix t/op/stat\.t failures on hurd
&nbsp;    DEBPKG&#8203;:fixes/nntp\_docs \- https://bugs.debian.org/51962 Net&#8203;::NNTP&#8203;: Correct innd/nnrpd confusion in relation to Reader option
&nbsp;    DEBPKG&#8203;:fixes/math\_complex\_doc\_great\_circle \- https://bugs.debian.org/697567 \[rt\.cpan\.org \#114104\] Math&#8203;::Trig&#8203;: clarify definition of great\_circle\_midpoint
&nbsp;    DEBPKG&#8203;:fixes/math\_complex\_doc\_see\_also \- https://bugs.debian.org/697568 \[rt\.cpan\.org \#114105\] Math&#8203;::Trig&#8203;: add missing SEE ALSO
&nbsp;    DEBPKG&#8203;:fixes/math\_complex\_doc\_angle\_units \- https://bugs.debian.org/731505 \[rt\.cpan\.org \#114106\] Math&#8203;::Trig&#8203;: document angle units
&nbsp;    DEBPKG&#8203;:fixes/cpan\_web\_link \- https://bugs.debian.org/367291 CPAN&#8203;: Add link to main CPAN web site
&nbsp;    DEBPKG&#8203;:fixes/time\_piece\_doc \- https://bugs.debian.org/817925 Time&#8203;::Piece&#8203;: Improve documentation for add\_months and add\_years
&nbsp;    DEBPKG&#8203;:fixes/perlbug\-refactor \- https://bugs.debian.org/822463 \[perl \#128020\] perlbug&#8203;: Refactor duplicated file reading code
&nbsp;    DEBPKG&#8203;:fixes/perlbug\-linewrap \- https://bugs.debian.org/822463 \[perl \#128020\] perlbug&#8203;: wrap overly long lines
&nbsp;    DEBPKG&#8203;:fixes/hurd\_sigaction \- https://bugs.debian.org/825016 \[d54f4ed\] ext/POSIX/t/sigaction\.t&#8203;: Skip uid and pid tests on GNU/Hurd
&nbsp;    DEBPKG&#8203;:fixes/hurd\_hints \- \[4694301\] https://bugs.debian.org/825020 \[perl \#128279\] Modify hints for Hurd per Debian ticket 825020\.
&nbsp;    DEBPKG&#8203;:fixes/extutils\-parsexs\-reproducibility \- \[perl \#128517\] https://bugs.debian.org/829296 Make the output of ExtUtils&#8203;::ParseXS reproducible
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/sitecustomize\-in\-etc \- Look for sitecustomize\.pl in /etc/perl rather than sitelib on Debian systems
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/test\-suite\-without\-dot \- \[perl \#127810\] Patch unit tests to explicitly insert "\." into @&#8203;INC when needed\.
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/eumm\-without\-dot \- \[perl \#127810\] Add PERL\_USE\_UNSAFE\_INC support to EU&#8203;::MM for fortify\_inc support\.
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/cpan\-without\-dot \- \[perl \#127810\] Set PERL\_USE\_UNSAFE\_INC for cpan usage
&nbsp;    DEBPKG&#8203;:debian/document\_inc\_removal \- Document in perlvar that we remove '\.' from @&#8203;INC by default
&nbsp;    DEBPKG&#8203;:fixes/extutils\_makemaker\_reproducible \- https&#8203;://bugs\.debian\.org/835815 https://bugs.debian.org/834190 Make perllocal\.pod files reproducible
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/remove\-inc\-test \- Remove test for '\.' in @&#8203;INC as it might not be
&nbsp;    DEBPKG&#8203;:debian/customized \- Update customized\.dat for files patched in Debian
&nbsp;    DEBPKG&#8203;:fixes/file\_path\_hurd\_errno \- File\-Path&#8203;: Fix test failure in Hurd due to hard\-coded ENOENT
&nbsp;    DEBPKG&#8203;:debian/hppa\_op\_optimize\_workaround \- https://bugs.debian.org/838613 Temporarily lower the optimization of op\.c on hppa due to gcc\-6 problems
&nbsp;    DEBPKG&#8203;:fixes/test\-builder\-warning \- https://bugs.debian.org/840968 Silence a 'used only once' warning in Test&#8203;::Builder
&nbsp;    DEBPKG&#8203;:fixes/longdblinf\-randomness \- \[dd68853\] \[perl \#130133\] https://bugs.debian.org/844752 Configure&#8203;: fix garbage filtering with 80\-bit long doubles
&nbsp;    DEBPKG&#8203;:debian/installman\-utf8 \- https://bugs.debian.org/840211 Generate man pages with UTF\-8 characters



---
@&#8203;INC for perl 5\.24\.1&#8203;:
&nbsp;    /etc/perl
&nbsp;    /usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1
&nbsp;    /usr/local/share/perl/5\.24\.1
&nbsp;    /usr/lib/x86\_64\-linux\-gnu/perl5/5\.24
&nbsp;    /usr/share/perl5
&nbsp;    /usr/lib/x86\_64\-linux\-gnu/perl/5\.24
&nbsp;    /usr/share/perl/5\.24
&nbsp;    /usr/local/lib/site\_perl
&nbsp;    /usr/lib/x86\_64\-linux\-gnu/perl\-base



---
Environment for perl 5\.24\.1&#8203;:
&nbsp;    HOME=/home/jdhedden
&nbsp;    LANG=C
&nbsp;    LANGUAGE=C
&nbsp;    LC\_ALL=C
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/home/jdhedden/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/games
&nbsp;    PERLIO=perlio
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @jdhedden

<details><summary>0001-Upgrade-to-Thread-Queue-3.12.patch</summary>

```diff
From 844313837a7de0aba2231e97d3a32e0f9831d695 Mon Sep 17 00:00:00 2001
From: jdhedden <jdhedden@cpan.org>
Date: Thu, 9 Feb 2017 13:43:19 -0500
Subject: [PATCH] Upgrade to Thread::Queue 3.12

---
 Porting/Maintainers.pl                |  2 +-
 dist/Thread-Queue/lib/Thread/Queue.pm | 24 +++++++++++++-----
 dist/Thread-Queue/t/01_basic.t        |  2 +-
 dist/Thread-Queue/t/02_refs.t         |  2 +-
 dist/Thread-Queue/t/03_peek.t         |  2 +-
 dist/Thread-Queue/t/05_extract.t      |  2 +-
 dist/Thread-Queue/t/06_insert.t       |  2 +-
 dist/Thread-Queue/t/07_lock.t         |  2 +-
 dist/Thread-Queue/t/10_timed.t        |  2 +-
 dist/Thread-Queue/t/11_limit.t        | 47 +++++++++++++++++++++++++++++++----
 10 files changed, 68 insertions(+), 19 deletions(-)

diff --git a/Porting/Maintainers.pl b/Porting/Maintainers.pl
index 9f6bf75d55..2198cf2208 100755
--- a/Porting/Maintainers.pl
+++ b/Porting/Maintainers.pl
@@ -1220,7 +1220,7 @@ use File::Glob qw(:case);
     # correct for this (and Thread::Semaphore, threads, and threads::shared)
     # to be under dist/ rather than cpan/
     'Thread::Queue' => {
-        'DISTRIBUTION' => 'JDHEDDEN/Thread-Queue-3.11.tar.gz',
+        'DISTRIBUTION' => 'JDHEDDEN/Thread-Queue-3.12.tar.gz',
         'FILES'        => q[dist/Thread-Queue],
         'EXCLUDED'     => [
             qr{^examples/},
diff --git a/dist/Thread-Queue/lib/Thread/Queue.pm b/dist/Thread-Queue/lib/Thread/Queue.pm
index 9f896b72ea..d8e47880bc 100644
--- a/dist/Thread-Queue/lib/Thread/Queue.pm
+++ b/dist/Thread-Queue/lib/Thread/Queue.pm
@@ -3,7 +3,7 @@ package Thread::Queue;
 use strict;
 use warnings;
 
-our $VERSION = '3.11';
+our $VERSION = '3.12';
 $VERSION = eval $VERSION;
 
 use threads::shared 1.21;
@@ -80,11 +80,13 @@ sub dequeue
 
     # Wait for requisite number of items
     cond_wait(%$self) while ((@$queue < $count) && ! $$self{'ENDED'});
-    cond_signal(%$self) if ((@$queue >= $count) || $$self{'ENDED'});
 
     # If no longer blocking, try getting whatever is left on the queue
     return $self->dequeue_nb($count) if ($$self{'ENDED'});
 
+    # Signal other threads that might be blocked on this queue
+    cond_signal(%$self) if ((@$queue >= $count) || $$self{'ENDED'});
+
     # Return single item
     return shift(@$queue) if ($count == 1);
 
@@ -103,6 +105,9 @@ sub dequeue_nb
 
     my $count = @_ ? $self->_validate_count(shift) : 1;
 
+    # Signal other threads that might be blocked on this queue
+    cond_signal(%$self) if ((@$queue >= $count) || $$self{'ENDED'});
+
     # Return single item
     return shift(@$queue) if ($count == 1);
 
@@ -135,7 +140,6 @@ sub dequeue_timed
     while ((@$queue < $count) && ! $$self{'ENDED'}) {
         last if (! cond_timedwait(%$self, $timeout));
     }
-    cond_signal(%$self) if ((@$queue >= $count) || $$self{'ENDED'});
 
     # Get whatever we need off the queue if available
     return $self->dequeue_nb($count);
@@ -263,14 +267,19 @@ sub _validate_count
     if (! defined($count) ||
         ! looks_like_number($count) ||
         (int($count) != $count) ||
-        ($count < 1))
+        ($count < 1) ||
+        ($$self{'LIMIT'} && $count > $$self{'LIMIT'}))
     {
         require Carp;
         my ($method) = (caller(1))[3];
         my $class_name = ref($self);
         $method =~ s/$class_name\:://;
         $count = 'undef' if (! defined($count));
-        Carp::croak("Invalid 'count' argument ($count) to '$method' method");
+        if ($$self{'LIMIT'} && $count > $$self{'LIMIT'}) {
+            Carp::croak("'count' argument ($count) to '$method' method exceeds queue size limit ($$self{'LIMIT'})");
+        } else {
+            Carp::croak("Invalid 'count' argument ($count) to '$method' method");
+        }
     }
 
     return $count;
@@ -304,7 +313,7 @@ Thread::Queue - Thread-safe queues
 
 =head1 VERSION
 
-This document describes Thread::Queue version 3.11
+This document describes Thread::Queue version 3.12
 
 =head1 SYNOPSIS
 
@@ -494,6 +503,9 @@ C<limit> does not prevent enqueuing items beyond that count:
                          # 'undef')
  $q->limit = 0;          # Queue size is now unlimited
 
+Calling any of the dequeue methods with C<COUNT> greater than a queue's
+C<limit> will generate an error.
+
 =item ->end()
 
 Declares that no more items will be added to the queue.
diff --git a/dist/Thread-Queue/t/01_basic.t b/dist/Thread-Queue/t/01_basic.t
index 4ec51957ae..2983f0b700 100644
--- a/dist/Thread-Queue/t/01_basic.t
+++ b/dist/Thread-Queue/t/01_basic.t
@@ -13,7 +13,7 @@ use threads;
 use Thread::Queue;
 
 if ($] == 5.008) {
-    require './t/test.pl';   # Test::More work-alike for Perl 5.8.0
+    require 't/test.pl';   # Test::More work-alike for Perl 5.8.0
 } else {
     require Test::More;
 }
diff --git a/dist/Thread-Queue/t/02_refs.t b/dist/Thread-Queue/t/02_refs.t
index fdf8f6bad2..0cebdc1db3 100644
--- a/dist/Thread-Queue/t/02_refs.t
+++ b/dist/Thread-Queue/t/02_refs.t
@@ -14,7 +14,7 @@ use threads::shared;
 use Thread::Queue;
 
 if ($] == 5.008) {
-    require './t/test.pl';   # Test::More work-alike for Perl 5.8.0
+    require 't/test.pl';   # Test::More work-alike for Perl 5.8.0
 } else {
     require Test::More;
 }
diff --git a/dist/Thread-Queue/t/03_peek.t b/dist/Thread-Queue/t/03_peek.t
index 29ef75e7fe..d543b59469 100644
--- a/dist/Thread-Queue/t/03_peek.t
+++ b/dist/Thread-Queue/t/03_peek.t
@@ -13,7 +13,7 @@ use threads;
 use Thread::Queue;
 
 if ($] == 5.008) {
-    require './t/test.pl';   # Test::More work-alike for Perl 5.8.0
+    require 't/test.pl';   # Test::More work-alike for Perl 5.8.0
 } else {
     require Test::More;
 }
diff --git a/dist/Thread-Queue/t/05_extract.t b/dist/Thread-Queue/t/05_extract.t
index d8cb417be9..de0e78bfd0 100644
--- a/dist/Thread-Queue/t/05_extract.t
+++ b/dist/Thread-Queue/t/05_extract.t
@@ -13,7 +13,7 @@ use threads;
 use Thread::Queue;
 
 if ($] == 5.008) {
-    require './t/test.pl';   # Test::More work-alike for Perl 5.8.0
+    require 't/test.pl';   # Test::More work-alike for Perl 5.8.0
 } else {
     require Test::More;
 }
diff --git a/dist/Thread-Queue/t/06_insert.t b/dist/Thread-Queue/t/06_insert.t
index 93617e13a3..4f9d1dff5e 100644
--- a/dist/Thread-Queue/t/06_insert.t
+++ b/dist/Thread-Queue/t/06_insert.t
@@ -13,7 +13,7 @@ use threads;
 use Thread::Queue;
 
 if ($] == 5.008) {
-    require './t/test.pl';   # Test::More work-alike for Perl 5.8.0
+    require 't/test.pl';   # Test::More work-alike for Perl 5.8.0
 } else {
     require Test::More;
 }
diff --git a/dist/Thread-Queue/t/07_lock.t b/dist/Thread-Queue/t/07_lock.t
index 633722103c..b20e0604ca 100644
--- a/dist/Thread-Queue/t/07_lock.t
+++ b/dist/Thread-Queue/t/07_lock.t
@@ -14,7 +14,7 @@ use Thread::Queue;
 use Thread::Semaphore;
 
 if ($] == 5.008) {
-    require './t/test.pl';   # Test::More work-alike for Perl 5.8.0
+    require 't/test.pl';   # Test::More work-alike for Perl 5.8.0
 } else {
     require Test::More;
 }
diff --git a/dist/Thread-Queue/t/10_timed.t b/dist/Thread-Queue/t/10_timed.t
index da8b03a7eb..8404720ed6 100644
--- a/dist/Thread-Queue/t/10_timed.t
+++ b/dist/Thread-Queue/t/10_timed.t
@@ -13,7 +13,7 @@ use threads;
 use Thread::Queue;
 
 if ($] == 5.008) {
-    require './t/test.pl';   # Test::More work-alike for Perl 5.8.0
+    require 't/test.pl';   # Test::More work-alike for Perl 5.8.0
 } else {
     require Test::More;
 }
diff --git a/dist/Thread-Queue/t/11_limit.t b/dist/Thread-Queue/t/11_limit.t
index 1bd88b39a1..56d813ffd1 100644
--- a/dist/Thread-Queue/t/11_limit.t
+++ b/dist/Thread-Queue/t/11_limit.t
@@ -19,7 +19,7 @@ use Thread::Queue;
 
 use Test::More;
 
-plan tests => 8;
+plan tests => 13;
 
 my $q = Thread::Queue->new();
 my $rpt = Thread::Queue->new();
@@ -82,12 +82,12 @@ $rpt->enqueue($q->pending);
 # q = (4, 5, 'foo'); r = (4, 3, 4, 3)
 
 # Read all items from queue
-my @item = $q->dequeue(3);
-is_deeply(\@item, [4, 5, 'foo'], 'Dequeued 3 items');
+my @items = $q->dequeue(3);
+is_deeply(\@items, [4, 5, 'foo'], 'Dequeued 3 items');
 # Thread is now unblocked
 
-@item = $q->dequeue(2);
-is_deeply(\@item, [6, 7], 'Dequeued 2 items');
+@items = $q->dequeue(2);
+is_deeply(\@items, [6, 7], 'Dequeued 2 items');
 
 # Thread is now unblocked
 # Handshake with thread
@@ -96,6 +96,43 @@ $rpt->enqueue('go');
 # (7) - Done
 $th->join;
 
+# It's an error to call dequeue methods with COUNT > LIMIT
+eval { $q->dequeue(5); };
+like($@, qr/exceeds queue size limit/, $@);
+
+# Bug #120157
+#  Fix deadlock from combination of dequeue_nb, enqueue and queue size limit
+
+# (1) Fill queue
+$q->enqueue(1..3);
+is($q->pending, 3, 'Queue loaded');
+
+$th = threads->create( sub {
+    $rpt->enqueue('go');
+
+    # (3) Thread blocks trying to add to full queue
+    $q->enqueue(99);
+
+    # (5) Thread exits
+    $rpt->enqueue('done');
+});
+
+# (2) Wait for thread to block on enqueue() call
+threads->yield();
+sleep(1);
+is($rpt->dequeue(), 'go', 'Thread blocked');
+
+# (4) Dequeue items - this will cause thread to unblock
+@items = ();
+while (my $item = $q->dequeue_nb()) {
+    push(@items, $item);
+}
+is_deeply(\@items, [1,2,3,99], 'Dequeued items');
+
+# (6) - Done
+$th->join();
+is($rpt->dequeue(), 'done', 'Thread exited');
+
 exit(0);
 
 # EOF
-- 
2.11.0


```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Thu\, 09 Feb 2017 18&#8203;:47&#8203;:22 GMT\, jdhedden@&#8203;cpan\.org wrote&#8203;:
> This is a bug report for perl from jdhedden@&#8203;cpan\.org\,
> generated with the help of perlbug 1\.40 running under perl 5\.24\.1\.
> 
> The attached patch upgrades 'Thread&#8203;::Queue' to version 3\.12\.
> 
> This is a fix from Chad Fox for
> \[rt\.cpan\.org \#120157\] deadlock using dequeue\_nb on size limited queue
> https://rt.cpan.org/Ticket/Display.html?id=120157
> 

When I applied this to Perl 5 blead \(f40abc997d245cb9516bc71ad192596d10097972\) and configured perl with '\-Dusethreads'\, I got the following failure&#8203;:

\#\#\#\#\#
 cd t;\./perl harness \-v \.\./dist/Thread\-Queue/t/11\_limit\.t; cd \-

ok 1 \- Queue limit set
ok 2 \- Dequeued item 1
ok 3 \- Queue count
ok 4 \- Dequeued item 2
ok 5 \- Dequeued item 3
ok 6 \- Dequeued 3 items
ok 7 \- Dequeued 2 items
ok 8 \- Queue reports
ok 9 \- 'count' argument \(5\) to 'dequeue' method exceeds queue size limit \(3\) at t/11\_limit\.t line 100\.
ok 10 \- Queue loaded
ok 11 \- Thread blocked
not ok 12 \- Dequeued items

\#   Failed test 'Dequeued items'
\#   at t/11\_limit\.t line 130\.
\#     Structures begin differing at&#8203;:
\#          $got\->\[3\] = Does not exist
\#     $expected\->\[3\] = '99'
ok 13 \- Thread exited
\# Looks like you failed 1 test of 13\.
Dubious\, test returned 1 \(wstat 256\, 0x100\)
Failed 1/13 subtests 

Test Summary Report


---
\.\./dist/Thread\-Queue/t/11\_limit\.t \(Wstat&#8203;: 256 Tests&#8203;: 13 Failed&#8203;: 1\)
&nbsp;  Failed test&#8203;:  12
&nbsp;  Non\-zero exit status&#8203;: 1
Files=1\, Tests=13\,  2 wallclock secs \( 0\.01 usr  0\.00 sys \+  0\.10 cusr  0\.00 csys =  0\.11 CPU\)
Result&#8203;: FAIL
\#\#\#\#\#





---
> Flags&#8203;:
>     category=library
>     severity=low
>     Type=Patch
>     PatchStatus=HasPatch
>     module=Thread&#8203;::Queue
> \-\-\-
> Site configuration information for perl 5\.24\.1&#8203;:
> 
> Configured by Debian Project at Sun Jan 15 23&#8203;:35&#8203;:20 UTC 2017\.
> 
> Summary of my perl5 \(revision 5 version 24 subversion 1\)
> configuration&#8203;:
> 
> Platform&#8203;:
>   osname=linux\, osvers=3\.16\.0\, archname=x86\_64\-linux\-gnu\-thread\-multi
>   uname='linux localhost 3\.16\.0 \#1 smp debian 3\.16\.0 x86\_64 gnulinux '
>   config\_args='\-Dusethreads \-Duselargefiles \-Dcc=x86\_64\-linux\-gnu\-gcc
> \-Dcpp=x86\_64\-linux\-gnu\-cpp \-Dld=x86\_64\-linux\-gnu\-gcc \-Dccflags=\-
> DDEBIAN \-Wdate\-time \-D\_FORTIFY\_SOURCE=2 \-g \-O2 \-fdebug\-prefix\-
> map=/build/perl\-eDFQz3/perl\-5\.24\.1=\. \-fstack\-protector\-strong \-Wformat
> \-Werror=format\-security \-Dldflags= \-Wl\,\-z\,relro \-Dlddlflags=\-shared
> \-Wl\,\-z\,relro \-Dcccdlflags=\-fPIC \-Darchname=x86\_64\-linux\-gnu
> \-Dprefix=/usr \-Dprivlib=/usr/share/perl/5\.24
> \-Darchlib=/usr/lib/x86\_64\-linux\-gnu/perl/5\.24 \-Dvendorprefix=/usr
> \-Dvendorlib=/usr/share/perl5 \-Dvendorarch=/usr/lib/x86\_64\-linux\-
> gnu/perl5/5\.24 \-Dsiteprefix=/usr/local
> \-Dsitelib=/usr/local/share/perl/5\.24\.1
> \-Dsitearch=/usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1
> \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/share/man/man3
> \-Dsiteman1dir=/usr/local/man/man1 \-Dsiteman3dir=/usr/local/man/man3
> \-Dusesitecustomize \-Duse64bitint \-Dman1ext=1 \-Dman3ext=3perl
> \-Dpager=/usr/bin/sensible\-pager \-Uafs \-Ud\_csh \-Ud\_ualarm \-Uusesfio
> \-Uusenm \-Ui\_libutil \-Uversiononly \-DDEBUGGING=\-g \-Doptimize=\-O2 \-dEs
> \-Duseshrplib \-Dlibperl=libperl\.so\.5\.24\.1'
>   hint=recommended\, useposix=true\, d\_sigaction=define
>   useithreads=define\, usemultiplicity=define
>   use64bitint=define\, use64bitall=define\, uselongdouble=undef
>   usemymalloc=n\, bincompat5005=undef
> Compiler&#8203;:
>   cc='x86\_64\-linux\-gnu\-gcc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE
> \-DDEBIAN \-fwrapv \-fno\-strict\-aliasing \-pipe \-I/usr/local/include
> \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
>   optimize='\-O2 \-g'\,
>   cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fwrapv \-fno\-strict\-
> aliasing \-pipe \-I/usr/local/include'
>   ccversion=''\, gccversion='6\.3\.0 20161229'\, gccosandvers=''
>   intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678\,
> doublekind=3
>   d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16\,
> longdblkind=3
>   ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> lseeksize=8
>   alignbytes=8\, prototype=define
> Linker and Libraries&#8203;:
>   ld='x86\_64\-linux\-gnu\-gcc'\, ldflags =' \-fstack\-protector\-strong
> \-L/usr/local/lib'
>   libpth=/usr/local/lib /usr/lib/gcc/x86\_64\-linux\-gnu/6/include\-fixed
> /usr/include/x86\_64\-linux\-gnu /usr/lib /lib/x86\_64\-linux\-gnu
> /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib
>   libs=\-lgdbm \-lgdbm\_compat \-ldb \-ldl \-lm \-lpthread \-lc \-lcrypt
>   perllibs=\-ldl \-lm \-lpthread \-lc \-lcrypt
>   libc=libc\-2\.24\.so\, so=so\, useshrplib=true\, libperl=libperl\.so\.5\.24
>   gnulibc\_version='2\.24'
> Dynamic Linking&#8203;:
>   dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
>   cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib \-fstack\-
> protector\-strong'
> 
> Locally applied patches&#8203;:
>     DEBPKG&#8203;:debian/cpan\_definstalldirs \- Provide a sensible INSTALLDIRS
> default for modules installed from CPAN\.
>     DEBPKG&#8203;:debian/db\_file\_ver \- https://bugs.debian.org/340047 Remove
> overly restrictive DB\_File version check\.
>     DEBPKG&#8203;:debian/doc\_info \- Replace generic man\(1\) instructions with
> Debian\-specific information\.
>     DEBPKG&#8203;:debian/enc2xs\_inc \- https://bugs.debian.org/290336 Tweak
> enc2xs to follow symlinks and ignore missing @&#8203;INC directories\.
>     DEBPKG&#8203;:debian/errno\_ver \- https://bugs.debian.org/343351 Remove
> Errno version check due to upgrade problems with long\-running
> processes\.
>     DEBPKG&#8203;:debian/libperl\_embed\_doc \- https://bugs.debian.org/186778
> Note that libperl\-dev package is required for embedded linking
>     DEBPKG&#8203;:fixes/respect\_umask \- Respect umask during installation
>     DEBPKG&#8203;:debian/writable\_site\_dirs \- Set umask approproately for
> site install directories
>     DEBPKG&#8203;:debian/extutils\_set\_libperl\_path \- EU&#8203;:MM&#8203;: set location of
> libperl\.a under /usr/lib
>     DEBPKG&#8203;:debian/no\_packlist\_perllocal \- Don't install \.packlist or
> perllocal\.pod for perl or vendor
>     DEBPKG&#8203;:debian/fakeroot \- Postpone LD\_LIBRARY\_PATH evaluation to
> the binary targets\.
>     DEBPKG&#8203;:debian/instmodsh\_doc \- Debian policy doesn't install
> \.packlist files for core or vendor\.
>     DEBPKG&#8203;:debian/ld\_run\_path \- Remove standard libs from LD\_RUN\_PATH
> as per Debian policy\.
>     DEBPKG&#8203;:debian/libnet\_config\_path \- Set location of libnet\.cfg to
> /etc/perl/Net as /usr may not be writable\.
>     DEBPKG&#8203;:debian/mod\_paths \- Tweak @&#8203;INC ordering for Debian
>     DEBPKG&#8203;:debian/prune\_libs \- https://bugs.debian.org/128355 Prune
> the list of libraries wanted to what we actually need\.
>     DEBPKG&#8203;:fixes/net\_smtp\_docs \- \[rt\.cpan\.org \#36038\]
> https://bugs.debian.org/100195 Document the Net&#8203;::SMTP 'Port' option
>     DEBPKG&#8203;:debian/perlivp \- https://bugs.debian.org/510895 Make
> perlivp skip include directories in /usr/local
>     DEBPKG&#8203;:debian/deprecate\-with\-apt \- https://bugs.debian.org/747628
> Point users to Debian packages of deprecated core modules
>     DEBPKG&#8203;:debian/squelch\-locale\-warnings \-
> https://bugs.debian.org/508764 Squelch locale warnings in Debian
> package maintainer scripts
>     DEBPKG&#8203;:debian/skip\-upstream\-git\-tests \- Skip tests specific to the
> upstream Git repository
>     DEBPKG&#8203;:debian/patchlevel \- https://bugs.debian.org/567489 List
> packaged patches for 5\.24\.1\-1 in patchlevel\.h
>     DEBPKG&#8203;:debian/skip\-kfreebsd\-crash \- https://bugs.debian.org/628493
> \[perl \#96272\] Skip a crashing test case in t/op/threads\.t on
> GNU/kFreeBSD
>     DEBPKG&#8203;:fixes/document\_makemaker\_ccflags \-
> https://bugs.debian.org/628522 \[rt\.cpan\.org \#68613\] Document that
> CCFLAGS should include $Config\{ccflags\}
>     DEBPKG&#8203;:debian/find\_html2text \- https://bugs.debian.org/640479
> Configure CPAN&#8203;::Distribution with correct name of html2text
>     DEBPKG&#8203;:debian/perl5db\-x\-terminal\-emulator\.patch \-
> https://bugs.debian.org/668490 Invoke x\-terminal\-emulator rather than
> xterm in perl5db\.pl
>     DEBPKG&#8203;:debian/cpan\-missing\-site\-dirs \-
> https://bugs.debian.org/688842 Fix CPAN&#8203;::FirstTime defaults with
> nonexisting site dirs if a parent is writable
>     DEBPKG&#8203;:fixes/memoize\_storable\_nstore \- \[rt\.cpan\.org \#77790\]
> https://bugs.debian.org/587650 Memoize&#8203;::Storable&#8203;: respect 'nstore'
> option not respected
>     DEBPKG&#8203;:debian/regen\-skip \- Skip a regeneration check in unrelated
> git repositories
>     DEBPKG&#8203;:debian/makemaker\-pasthru \- https://bugs.debian.org/758471
> Pass LD settings through to subdirectories
>     DEBPKG&#8203;:debian/makemaker\-manext \- https://bugs.debian.org/247370
> Make EU&#8203;::MakeMaker honour MANnEXT settings in generated manpage
> headers
>     DEBPKG&#8203;:debian/devel\-ppport\-reproducibility \-
> https://bugs.debian.org/801523 Sort the list of XS code files when
> generating RealPPPort\.xs
>     DEBPKG&#8203;:debian/encode\-unicode\-bom\-doc \-
> https://bugs.debian.org/798727 Document Debian backport of
> Encode&#8203;::Unicode fix
>     DEBPKG&#8203;:debian/kfreebsd\-softupdates \-
> https://bugs.debian.org/796798 Work around Debian Bug\#796798
>     DEBPKG&#8203;:fixes/autodie\-scope \- https://bugs.debian.org/798096 Fix a
> scoping issue with "no autodie" and the "system" sub
>     DEBPKG&#8203;:fixes/crosscompile\-no\-targethost \- \[23695c0\] \[perl \#127234\]
> Fix the Configure escape with usecrosscompile but no targethost
>     DEBPKG&#8203;:fixes/memoize\-pod \- \[rt\.cpan\.org \#89441\] Fix POD errors in
> Memoize
>     DEBPKG&#8203;:fixes/ok\-pod \- Added encoding for pod\.
>     DEBPKG&#8203;:debian/hurd\-softupdates \- https://bugs.debian.org/822735
> Fix t/op/stat\.t failures on hurd
>     DEBPKG&#8203;:fixes/nntp\_docs \- https://bugs.debian.org/51962 Net&#8203;::NNTP&#8203;:
> Correct innd/nnrpd confusion in relation to Reader option
>     DEBPKG&#8203;:fixes/math\_complex\_doc\_great\_circle \-
> https://bugs.debian.org/697567 \[rt\.cpan\.org \#114104\] Math&#8203;::Trig&#8203;:
> clarify definition of great\_circle\_midpoint
>     DEBPKG&#8203;:fixes/math\_complex\_doc\_see\_also \-
> https://bugs.debian.org/697568 \[rt\.cpan\.org \#114105\] Math&#8203;::Trig&#8203;: add
> missing SEE ALSO
>     DEBPKG&#8203;:fixes/math\_complex\_doc\_angle\_units \-
> https://bugs.debian.org/731505 \[rt\.cpan\.org \#114106\] Math&#8203;::Trig&#8203;:
> document angle units
>     DEBPKG&#8203;:fixes/cpan\_web\_link \- https://bugs.debian.org/367291 CPAN&#8203;:
> Add link to main CPAN web site
>     DEBPKG&#8203;:fixes/time\_piece\_doc \- https://bugs.debian.org/817925
> Time&#8203;::Piece&#8203;: Improve documentation for add\_months and add\_years
>     DEBPKG&#8203;:fixes/perlbug\-refactor \- https://bugs.debian.org/822463
> \[perl \#128020\] perlbug&#8203;: Refactor duplicated file reading code
>     DEBPKG&#8203;:fixes/perlbug\-linewrap \- https://bugs.debian.org/822463
> \[perl \#128020\] perlbug&#8203;: wrap overly long lines
>     DEBPKG&#8203;:fixes/hurd\_sigaction \- https://bugs.debian.org/825016
> \[d54f4ed\] ext/POSIX/t/sigaction\.t&#8203;: Skip uid and pid tests on GNU/Hurd
>     DEBPKG&#8203;:fixes/hurd\_hints \- \[4694301\] https://bugs.debian.org/825020
> \[perl \#128279\] Modify hints for Hurd per Debian ticket 825020\.
>     DEBPKG&#8203;:fixes/extutils\-parsexs\-reproducibility \- \[perl \#128517\]
> https://bugs.debian.org/829296 Make the output of ExtUtils&#8203;::ParseXS
> reproducible
>     DEBPKG&#8203;:debian/CVE\-2016\-1238/sitecustomize\-in\-etc \- Look for
> sitecustomize\.pl in /etc/perl rather than sitelib on Debian systems
>     DEBPKG&#8203;:debian/CVE\-2016\-1238/test\-suite\-without\-dot \- \[perl
> \#127810\] Patch unit tests to explicitly insert "\." into @&#8203;INC when
> needed\.
>     DEBPKG&#8203;:debian/CVE\-2016\-1238/eumm\-without\-dot \- \[perl \#127810\] Add
> PERL\_USE\_UNSAFE\_INC support to EU&#8203;::MM for fortify\_inc support\.
>     DEBPKG&#8203;:debian/CVE\-2016\-1238/cpan\-without\-dot \- \[perl \#127810\] Set
> PERL\_USE\_UNSAFE\_INC for cpan usage
>     DEBPKG&#8203;:debian/document\_inc\_removal \- Document in perlvar that we
> remove '\.' from @&#8203;INC by default
>     DEBPKG&#8203;:fixes/extutils\_makemaker\_reproducible \-
> https&#8203;://bugs\.debian\.org/835815 https://bugs.debian.org/834190 Make
> perllocal\.pod files reproducible
>     DEBPKG&#8203;:debian/CVE\-2016\-1238/remove\-inc\-test \- Remove test for '\.'
> in @&#8203;INC as it might not be
>     DEBPKG&#8203;:debian/customized \- Update customized\.dat for files patched
> in Debian
>     DEBPKG&#8203;:fixes/file\_path\_hurd\_errno \- File\-Path&#8203;: Fix test failure in
> Hurd due to hard\-coded ENOENT
>     DEBPKG&#8203;:debian/hppa\_op\_optimize\_workaround \-
> https://bugs.debian.org/838613 Temporarily lower the optimization of
> op\.c on hppa due to gcc\-6 problems
>     DEBPKG&#8203;:fixes/test\-builder\-warning \- https://bugs.debian.org/840968
> Silence a 'used only once' warning in Test&#8203;::Builder
>     DEBPKG&#8203;:fixes/longdblinf\-randomness \- \[dd68853\] \[perl \#130133\]
> https://bugs.debian.org/844752 Configure&#8203;: fix garbage filtering with
> 80\-bit long doubles
>     DEBPKG&#8203;:debian/installman\-utf8 \- https://bugs.debian.org/840211
> Generate man pages with UTF\-8 characters
> 
> \-\-\-
> @&#8203;INC for perl 5\.24\.1&#8203;:
>     /etc/perl
>     /usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1
>     /usr/local/share/perl/5\.24\.1
>     /usr/lib/x86\_64\-linux\-gnu/perl5/5\.24
>     /usr/share/perl5
>     /usr/lib/x86\_64\-linux\-gnu/perl/5\.24
>     /usr/share/perl/5\.24
>     /usr/local/lib/site\_perl
>     /usr/lib/x86\_64\-linux\-gnu/perl\-base
> 
> \-\-\-
> Environment for perl 5\.24\.1&#8203;:
>     HOME=/home/jdhedden
>     LANG=C
>     LANGUAGE=C
>     LC\_ALL=C
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
>     PATH=/home/jdhedden/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/games
>     PERLIO=perlio
>     PERL\_BADLANG \(unset\)
>     SHELL=/bin/bash


\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

On Mon\, 13 Feb 2017 13&#8203;:55&#8203;:32 GMT\, jkeenan wrote&#8203;:
> On Thu\, 09 Feb 2017 18&#8203;:47&#8203;:22 GMT\, jdhedden@&#8203;cpan\.org wrote&#8203;:
> > This is a bug report for perl from jdhedden@&#8203;cpan\.org\,
> > generated with the help of perlbug 1\.40 running under perl 5\.24\.1\.
> >
> > The attached patch upgrades 'Thread&#8203;::Queue' to version 3\.12\.
> >
> > This is a fix from Chad Fox for
> > \[rt\.cpan\.org \#120157\] deadlock using dequeue\_nb on size limited queue
> > https://rt.cpan.org/Ticket/Display.html?id=120157
> >
> 
> When I applied this to Perl 5 blead
> \(f40abc997d245cb9516bc71ad192596d10097972\) and configured perl with '\-
> Dusethreads'\, I got the following failure&#8203;:
> 
> \#\#\#\#\#
> cd t;\./perl harness \-v \.\./dist/Thread\-Queue/t/11\_limit\.t; cd \-
> 
> ok 1 \- Queue limit set
> ok 2 \- Dequeued item 1
> ok 3 \- Queue count
> ok 4 \- Dequeued item 2
> ok 5 \- Dequeued item 3
> ok 6 \- Dequeued 3 items
> ok 7 \- Dequeued 2 items
> ok 8 \- Queue reports
> ok 9 \- 'count' argument \(5\) to 'dequeue' method exceeds queue size
> limit \(3\) at t/11\_limit\.t line 100\.
> ok 10 \- Queue loaded
> ok 11 \- Thread blocked
> not ok 12 \- Dequeued items
> 
> \#   Failed test 'Dequeued items'
> \#   at t/11\_limit\.t line 130\.
> \#     Structures begin differing at&#8203;:
> \#          $got\->\[3\] = Does not exist
> \#     $expected\->\[3\] = '99'
> ok 13 \- Thread exited
> \# Looks like you failed 1 test of 13\.
> Dubious\, test returned 1 \(wstat 256\, 0x100\)
>  Failed 1/13 subtests
> 
> Test Summary Report
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \.\./dist/Thread\-Queue/t/11\_limit\.t \(Wstat&#8203;: 256 Tests&#8203;: 13 Failed&#8203;: 1\)
>   Failed test&#8203;:  12
>   Non\-zero exit status&#8203;: 1
> Files=1\, Tests=13\,  2 wallclock secs \( 0\.01 usr  0\.00 sys \+  0\.10 cusr
> 0\.00 csys =  0\.11 CPU\)
> Result&#8203;: FAIL
> \#\#\#\#\#
> 
> 

To make matters more confusing\, I got the FAIL above on Linux\, but I got a PASS on FreeBSD\-10\.3 where I customarily configure with&#8203;:

\#\#\#\#\#
sh \./Configure \-des \-Dusedevel \\
&nbsp;            \-Duseithreads \\
&nbsp;            \-Doptimize="\-O2 \-pipe \-fstack\-protector \-fno\-strict\-aliasing"
\#\#\#\#\#

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @jdhedden

On Mon\, 13 Feb 2017 05&#8203;:55&#8203;:32 \-0800\, jkeenan wrote&#8203;:
> On Thu\, 09 Feb 2017 18&#8203;:47&#8203;:22 GMT\, jdhedden@&#8203;cpan\.org wrote&#8203;:
> > This is a bug report for perl from jdhedden@&#8203;cpan\.org\,
> > generated with the help of perlbug 1\.40 running under perl 5\.24\.1\.
> >
> > The attached patch upgrades 'Thread&#8203;::Queue' to version 3\.12\.
> >
> > This is a fix from Chad Fox for
> > \[rt\.cpan\.org \#120157\] deadlock using dequeue\_nb on size limited queue
> > https://rt.cpan.org/Ticket/Display.html?id=120157
> >
> 
> When I applied this to Perl 5 blead
> \(f40abc997d245cb9516bc71ad192596d10097972\) and configured perl with '\-
> Dusethreads'\, I got the following failure&#8203;:
> 
> \#\#\#\#\#
> cd t;\./perl harness \-v \.\./dist/Thread\-Queue/t/11\_limit\.t; cd \-
> 
> ok 1 \- Queue limit set
> ok 2 \- Dequeued item 1
> ok 3 \- Queue count
> ok 4 \- Dequeued item 2
> ok 5 \- Dequeued item 3
> ok 6 \- Dequeued 3 items
> ok 7 \- Dequeued 2 items
> ok 8 \- Queue reports
> ok 9 \- 'count' argument \(5\) to 'dequeue' method exceeds queue size
> limit \(3\) at t/11\_limit\.t line 100\.
> ok 10 \- Queue loaded
> ok 11 \- Thread blocked
> not ok 12 \- Dequeued items
> 
> \#   Failed test 'Dequeued items'
> \#   at t/11\_limit\.t line 130\.
> \#     Structures begin differing at&#8203;:
> \#          $got\->\[3\] = Does not exist
> \#     $expected\->\[3\] = '99'
> ok 13 \- Thread exited
> \# Looks like you failed 1 test of 13\.
> Dubious\, test returned 1 \(wstat 256\, 0x100\)
>  Failed 1/13 subtests
> 
> Test Summary Report
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \.\./dist/Thread\-Queue/t/11\_limit\.t \(Wstat&#8203;: 256 Tests&#8203;: 13 Failed&#8203;: 1\)
>   Failed test&#8203;:  12
>   Non\-zero exit status&#8203;: 1
> Files=1\, Tests=13\,  2 wallclock secs \( 0\.01 usr  0\.00 sys \+  0\.10 cusr
> 0\.00 csys =  0\.11 CPU\)
> Result&#8203;: FAIL
> \#\#\#\#\#
> 
> 
> 
> \-\-\-
> > Flags&#8203;:
> >     category=library
> >     severity=low
> >     Type=Patch
> >     PatchStatus=HasPatch
> >     module=Thread&#8203;::Queue
> > \-\-\-
> > Site configuration information for perl 5\.24\.1&#8203;:
> >
> > Configured by Debian Project at Sun Jan 15 23&#8203;:35&#8203;:20 UTC 2017\.
> >
> > Summary of my perl5 \(revision 5 version 24 subversion 1\)
> > configuration&#8203;:
> >
> > Platform&#8203;:
> >   osname=linux\, osvers=3\.16\.0\, archname=x86\_64\-linux\-gnu\-thread\-multi
> >   uname='linux localhost 3\.16\.0 \#1 smp debian 3\.16\.0 x86\_64 gnulinux
> > '
> >   config\_args='\-Dusethreads \-Duselargefiles \-Dcc=x86\_64\-linux\-gnu\-gcc
> > \-Dcpp=x86\_64\-linux\-gnu\-cpp \-Dld=x86\_64\-linux\-gnu\-gcc \-Dccflags=\-
> > DDEBIAN \-Wdate\-time \-D\_FORTIFY\_SOURCE=2 \-g \-O2 \-fdebug\-prefix\-
> > map=/build/perl\-eDFQz3/perl\-5\.24\.1=\. \-fstack\-protector\-strong
> > \-Wformat
> > \-Werror=format\-security \-Dldflags= \-Wl\,\-z\,relro \-Dlddlflags=\-shared
> > \-Wl\,\-z\,relro \-Dcccdlflags=\-fPIC \-Darchname=x86\_64\-linux\-gnu
> > \-Dprefix=/usr \-Dprivlib=/usr/share/perl/5\.24
> > \-Darchlib=/usr/lib/x86\_64\-linux\-gnu/perl/5\.24 \-Dvendorprefix=/usr
> > \-Dvendorlib=/usr/share/perl5 \-Dvendorarch=/usr/lib/x86\_64\-linux\-
> > gnu/perl5/5\.24 \-Dsiteprefix=/usr/local
> > \-Dsitelib=/usr/local/share/perl/5\.24\.1
> > \-Dsitearch=/usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1
> > \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/share/man/man3
> > \-Dsiteman1dir=/usr/local/man/man1 \-Dsiteman3dir=/usr/local/man/man3
> > \-Dusesitecustomize \-Duse64bitint \-Dman1ext=1 \-Dman3ext=3perl
> > \-Dpager=/usr/bin/sensible\-pager \-Uafs \-Ud\_csh \-Ud\_ualarm \-Uusesfio
> > \-Uusenm \-Ui\_libutil \-Uversiononly \-DDEBUGGING=\-g \-Doptimize=\-O2 \-dEs
> > \-Duseshrplib \-Dlibperl=libperl\.so\.5\.24\.1'
> >   hint=recommended\, useposix=true\, d\_sigaction=define
> >   useithreads=define\, usemultiplicity=define
> >   use64bitint=define\, use64bitall=define\, uselongdouble=undef
> >   usemymalloc=n\, bincompat5005=undef
> > Compiler&#8203;:
> >   cc='x86\_64\-linux\-gnu\-gcc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE
> > \-DDEBIAN \-fwrapv \-fno\-strict\-aliasing \-pipe \-I/usr/local/include
> > \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
> >   optimize='\-O2 \-g'\,
> >   cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fwrapv \-fno\-strict\-
> > aliasing \-pipe \-I/usr/local/include'
> >   ccversion=''\, gccversion='6\.3\.0 20161229'\, gccosandvers=''
> >   intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678\,
> > doublekind=3
> >   d\_longlong=define\, longlongsize=8\, d\_longdbl=define\,
> > longdblsize=16\,
> > longdblkind=3
> >   ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> > lseeksize=8
> >   alignbytes=8\, prototype=define
> > Linker and Libraries&#8203;:
> >   ld='x86\_64\-linux\-gnu\-gcc'\, ldflags =' \-fstack\-protector\-strong
> > \-L/usr/local/lib'
> >   libpth=/usr/local/lib /usr/lib/gcc/x86\_64\-linux\-gnu/6/include\-fixed
> > /usr/include/x86\_64\-linux\-gnu /usr/lib /lib/x86\_64\-linux\-gnu
> > /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib
> >   libs=\-lgdbm \-lgdbm\_compat \-ldb \-ldl \-lm \-lpthread \-lc \-lcrypt
> >   perllibs=\-ldl \-lm \-lpthread \-lc \-lcrypt
> >   libc=libc\-2\.24\.so\, so=so\, useshrplib=true\, libperl=libperl\.so\.5\.24
> >   gnulibc\_version='2\.24'
> > Dynamic Linking&#8203;:
> >   dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
> >   cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib \-fstack\-
> > protector\-strong'
> >
> > Locally applied patches&#8203;:
> >     DEBPKG&#8203;:debian/cpan\_definstalldirs \- Provide a sensible
> > INSTALLDIRS
> > default for modules installed from CPAN\.
> >     DEBPKG&#8203;:debian/db\_file\_ver \- https://bugs.debian.org/340047 Remove
> > overly restrictive DB\_File version check\.
> >     DEBPKG&#8203;:debian/doc\_info \- Replace generic man\(1\) instructions with
> > Debian\-specific information\.
> >     DEBPKG&#8203;:debian/enc2xs\_inc \- https://bugs.debian.org/290336 Tweak
> > enc2xs to follow symlinks and ignore missing @&#8203;INC directories\.
> >     DEBPKG&#8203;:debian/errno\_ver \- https://bugs.debian.org/343351 Remove
> > Errno version check due to upgrade problems with long\-running
> > processes\.
> >     DEBPKG&#8203;:debian/libperl\_embed\_doc \- https://bugs.debian.org/186778
> > Note that libperl\-dev package is required for embedded linking
> >     DEBPKG&#8203;:fixes/respect\_umask \- Respect umask during installation
> >     DEBPKG&#8203;:debian/writable\_site\_dirs \- Set umask approproately for
> > site install directories
> >     DEBPKG&#8203;:debian/extutils\_set\_libperl\_path \- EU&#8203;:MM&#8203;: set location of
> > libperl\.a under /usr/lib
> >     DEBPKG&#8203;:debian/no\_packlist\_perllocal \- Don't install \.packlist or
> > perllocal\.pod for perl or vendor
> >     DEBPKG&#8203;:debian/fakeroot \- Postpone LD\_LIBRARY\_PATH evaluation to
> > the binary targets\.
> >     DEBPKG&#8203;:debian/instmodsh\_doc \- Debian policy doesn't install
> > \.packlist files for core or vendor\.
> >     DEBPKG&#8203;:debian/ld\_run\_path \- Remove standard libs from LD\_RUN\_PATH
> > as per Debian policy\.
> >     DEBPKG&#8203;:debian/libnet\_config\_path \- Set location of libnet\.cfg to
> > /etc/perl/Net as /usr may not be writable\.
> >     DEBPKG&#8203;:debian/mod\_paths \- Tweak @&#8203;INC ordering for Debian
> >     DEBPKG&#8203;:debian/prune\_libs \- https://bugs.debian.org/128355 Prune
> > the list of libraries wanted to what we actually need\.
> >     DEBPKG&#8203;:fixes/net\_smtp\_docs \- \[rt\.cpan\.org \#36038\]
> > https://bugs.debian.org/100195 Document the Net&#8203;::SMTP 'Port' option
> >     DEBPKG&#8203;:debian/perlivp \- https://bugs.debian.org/510895 Make
> > perlivp skip include directories in /usr/local
> >     DEBPKG&#8203;:debian/deprecate\-with\-apt \- https://bugs.debian.org/747628
> > Point users to Debian packages of deprecated core modules
> >     DEBPKG&#8203;:debian/squelch\-locale\-warnings \-
> > https://bugs.debian.org/508764 Squelch locale warnings in Debian
> > package maintainer scripts
> >     DEBPKG&#8203;:debian/skip\-upstream\-git\-tests \- Skip tests specific to
> > the
> > upstream Git repository
> >     DEBPKG&#8203;:debian/patchlevel \- https://bugs.debian.org/567489 List
> > packaged patches for 5\.24\.1\-1 in patchlevel\.h
> >     DEBPKG&#8203;:debian/skip\-kfreebsd\-crash \-
> > https://bugs.debian.org/628493
> > \[perl \#96272\] Skip a crashing test case in t/op/threads\.t on
> > GNU/kFreeBSD
> >     DEBPKG&#8203;:fixes/document\_makemaker\_ccflags \-
> > https://bugs.debian.org/628522 \[rt\.cpan\.org \#68613\] Document that
> > CCFLAGS should include $Config\{ccflags\}
> >     DEBPKG&#8203;:debian/find\_html2text \- https://bugs.debian.org/640479
> > Configure CPAN&#8203;::Distribution with correct name of html2text
> >     DEBPKG&#8203;:debian/perl5db\-x\-terminal\-emulator\.patch \-
> > https://bugs.debian.org/668490 Invoke x\-terminal\-emulator rather than
> > xterm in perl5db\.pl
> >     DEBPKG&#8203;:debian/cpan\-missing\-site\-dirs \-
> > https://bugs.debian.org/688842 Fix CPAN&#8203;::FirstTime defaults with
> > nonexisting site dirs if a parent is writable
> >     DEBPKG&#8203;:fixes/memoize\_storable\_nstore \- \[rt\.cpan\.org \#77790\]
> > https://bugs.debian.org/587650 Memoize&#8203;::Storable&#8203;: respect 'nstore'
> > option not respected
> >     DEBPKG&#8203;:debian/regen\-skip \- Skip a regeneration check in unrelated
> > git repositories
> >     DEBPKG&#8203;:debian/makemaker\-pasthru \- https://bugs.debian.org/758471
> > Pass LD settings through to subdirectories
> >     DEBPKG&#8203;:debian/makemaker\-manext \- https://bugs.debian.org/247370
> > Make EU&#8203;::MakeMaker honour MANnEXT settings in generated manpage
> > headers
> >     DEBPKG&#8203;:debian/devel\-ppport\-reproducibility \-
> > https://bugs.debian.org/801523 Sort the list of XS code files when
> > generating RealPPPort\.xs
> >     DEBPKG&#8203;:debian/encode\-unicode\-bom\-doc \-
> > https://bugs.debian.org/798727 Document Debian backport of
> > Encode&#8203;::Unicode fix
> >     DEBPKG&#8203;:debian/kfreebsd\-softupdates \-
> > https://bugs.debian.org/796798 Work around Debian Bug\#796798
> >     DEBPKG&#8203;:fixes/autodie\-scope \- https://bugs.debian.org/798096 Fix a
> > scoping issue with "no autodie" and the "system" sub
> >     DEBPKG&#8203;:fixes/crosscompile\-no\-targethost \- \[23695c0\] \[perl
> > \#127234\]
> > Fix the Configure escape with usecrosscompile but no targethost
> >     DEBPKG&#8203;:fixes/memoize\-pod \- \[rt\.cpan\.org \#89441\] Fix POD errors in
> > Memoize
> >     DEBPKG&#8203;:fixes/ok\-pod \- Added encoding for pod\.
> >     DEBPKG&#8203;:debian/hurd\-softupdates \- https://bugs.debian.org/822735
> > Fix t/op/stat\.t failures on hurd
> >     DEBPKG&#8203;:fixes/nntp\_docs \- https://bugs.debian.org/51962 Net&#8203;::NNTP&#8203;:
> > Correct innd/nnrpd confusion in relation to Reader option
> >     DEBPKG&#8203;:fixes/math\_complex\_doc\_great\_circle \-
> > https://bugs.debian.org/697567 \[rt\.cpan\.org \#114104\] Math&#8203;::Trig&#8203;:
> > clarify definition of great\_circle\_midpoint
> >     DEBPKG&#8203;:fixes/math\_complex\_doc\_see\_also \-
> > https://bugs.debian.org/697568 \[rt\.cpan\.org \#114105\] Math&#8203;::Trig&#8203;: add
> > missing SEE ALSO
> >     DEBPKG&#8203;:fixes/math\_complex\_doc\_angle\_units \-
> > https://bugs.debian.org/731505 \[rt\.cpan\.org \#114106\] Math&#8203;::Trig&#8203;:
> > document angle units
> >     DEBPKG&#8203;:fixes/cpan\_web\_link \- https://bugs.debian.org/367291 CPAN&#8203;:
> > Add link to main CPAN web site
> >     DEBPKG&#8203;:fixes/time\_piece\_doc \- https://bugs.debian.org/817925
> > Time&#8203;::Piece&#8203;: Improve documentation for add\_months and add\_years
> >     DEBPKG&#8203;:fixes/perlbug\-refactor \- https://bugs.debian.org/822463
> > \[perl \#128020\] perlbug&#8203;: Refactor duplicated file reading code
> >     DEBPKG&#8203;:fixes/perlbug\-linewrap \- https://bugs.debian.org/822463
> > \[perl \#128020\] perlbug&#8203;: wrap overly long lines
> >     DEBPKG&#8203;:fixes/hurd\_sigaction \- https://bugs.debian.org/825016
> > \[d54f4ed\] ext/POSIX/t/sigaction\.t&#8203;: Skip uid and pid tests on GNU/Hurd
> >     DEBPKG&#8203;:fixes/hurd\_hints \- \[4694301\]
> > https://bugs.debian.org/825020
> > \[perl \#128279\] Modify hints for Hurd per Debian ticket 825020\.
> >     DEBPKG&#8203;:fixes/extutils\-parsexs\-reproducibility \- \[perl \#128517\]
> > https://bugs.debian.org/829296 Make the output of ExtUtils&#8203;::ParseXS
> > reproducible
> >     DEBPKG&#8203;:debian/CVE\-2016\-1238/sitecustomize\-in\-etc \- Look for
> > sitecustomize\.pl in /etc/perl rather than sitelib on Debian systems
> >     DEBPKG&#8203;:debian/CVE\-2016\-1238/test\-suite\-without\-dot \- \[perl
> > \#127810\] Patch unit tests to explicitly insert "\." into @&#8203;INC when
> > needed\.
> >     DEBPKG&#8203;:debian/CVE\-2016\-1238/eumm\-without\-dot \- \[perl \#127810\] Add
> > PERL\_USE\_UNSAFE\_INC support to EU&#8203;::MM for fortify\_inc support\.
> >     DEBPKG&#8203;:debian/CVE\-2016\-1238/cpan\-without\-dot \- \[perl \#127810\] Set
> > PERL\_USE\_UNSAFE\_INC for cpan usage
> >     DEBPKG&#8203;:debian/document\_inc\_removal \- Document in perlvar that we
> > remove '\.' from @&#8203;INC by default
> >     DEBPKG&#8203;:fixes/extutils\_makemaker\_reproducible \-
> > https&#8203;://bugs\.debian\.org/835815 https://bugs.debian.org/834190 Make
> > perllocal\.pod files reproducible
> >     DEBPKG&#8203;:debian/CVE\-2016\-1238/remove\-inc\-test \- Remove test for '\.'
> > in @&#8203;INC as it might not be
> >     DEBPKG&#8203;:debian/customized \- Update customized\.dat for files
> > patched
> > in Debian
> >     DEBPKG&#8203;:fixes/file\_path\_hurd\_errno \- File\-Path&#8203;: Fix test failure
> > in
> > Hurd due to hard\-coded ENOENT
> >     DEBPKG&#8203;:debian/hppa\_op\_optimize\_workaround \-
> > https://bugs.debian.org/838613 Temporarily lower the optimization of
> > op\.c on hppa due to gcc\-6 problems
> >     DEBPKG&#8203;:fixes/test\-builder\-warning \-
> > https://bugs.debian.org/840968
> > Silence a 'used only once' warning in Test&#8203;::Builder
> >     DEBPKG&#8203;:fixes/longdblinf\-randomness \- \[dd68853\] \[perl \#130133\]
> > https://bugs.debian.org/844752 Configure&#8203;: fix garbage filtering with
> > 80\-bit long doubles
> >     DEBPKG&#8203;:debian/installman\-utf8 \- https://bugs.debian.org/840211
> > Generate man pages with UTF\-8 characters
> >
> > \-\-\-
> > @&#8203;INC for perl 5\.24\.1&#8203;:
> >     /etc/perl
> >     /usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1
> >     /usr/local/share/perl/5\.24\.1
> >     /usr/lib/x86\_64\-linux\-gnu/perl5/5\.24
> >     /usr/share/perl5
> >     /usr/lib/x86\_64\-linux\-gnu/perl/5\.24
> >     /usr/share/perl/5\.24
> >     /usr/local/lib/site\_perl
> >     /usr/lib/x86\_64\-linux\-gnu/perl\-base
> >
> > \-\-\-
> > Environment for perl 5\.24\.1&#8203;:
> >     HOME=/home/jdhedden
> >     LANG=C
> >     LANGUAGE=C
> >     LC\_ALL=C
> >     LD\_LIBRARY\_PATH \(unset\)
> >     LOGDIR \(unset\)
> >     PATH=/home/jdhedden/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/games
> >     PERLIO=perlio
> >     PERL\_BADLANG \(unset\)
> >     SHELL=/bin/bash


Thanks for this\.  I believe I have identified a race condition that requires additional code changes\.

Please close this ticket and I'll resubmit later\.

**Comment by p5pRT** [comment]
### From @jkeenan

OP will resubmit in new ticket\.


\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'rejected'
